AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/parameter/LocationActivity.java

```diff
@@ -0,0 +1,182 @@
+        inputSelect = SELECT_PICK_UP;
+        if (Objects.requireNonNull(pickUpInput.getText()).toString().isEmpty()) {
+          errorHint();
+        } else {
+          input = pickUpInput.getText().toString();
+          getAddressList(input);
+        }
+      } else {
+        inputSelect = SELECT_DROP_OFF;
+        if (Objects.requireNonNull(dropOffInput.getText()).toString().isEmpty()) {
+          errorHint();
+        } else {
+          input = dropOffInput.getText().toString();
+          getAddressList(input);
+        }
+      }
+    }
+  };
+  private Address pickUpAddress = null, dropOffAddress = null;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_select_location);
+    initData();
+    initView();
+  }
+
+  @Override
+  protected void initData() {
+
+  }
+
+  @Override
+  protected void initView() {
+    super.initView();
+    getSupportActionBar().setTitle(TAG);
+    addressListView = findViewById(R.id.address_list);
+    DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(addressListView.getContext(),
+        new LinearLayoutManager(LocationActivity.this).getOrientation());
+    addressListView.addItemDecoration(dividerItemDecoration);
+
+    addressListView.setAdapter(adapter);
+    pickUpInput = findViewById(R.id.pick_up_text);
+    dropOffInput = findViewById(R.id.drop_off_text);
+    Button searchPickUp = findViewById(R.id.search_pick_up);
+    Button searchDropOff = findViewById(R.id.search_drop_off);
+    searchPickUp.setOnClickListener(clickListener);
+    searchDropOff.setOnClickListener(clickListener);
+    mHandlerThread = new Handler(Looper.getMainLooper()) {
+      @Override
+      public void handleMessage(@NonNull Message msg) {
+        if (msg.what == UPDATE) {
+          adapter.notifyDataSetChanged();
+        } else if (msg.what == ERROR) {
+          errorHint();
+        }
+      }
+    };
+  }
+
+  private void errorHint() {
+    Utils.showMsg(ERROR_NO_PLACE, LocationActivity.this);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.location_menu, menu);
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+    if (item.getItemId() == R.id.save) {
+      checkInput();
+      return true;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  private void checkInput() {
+    if (pickUpAddress == null || dropOffAddress == null) {
+      errorHint();
+      return;
+    }
+    Intent intent = new Intent();
+    intent.putExtra(Constant.PICK_UP_LATITUDE, pickUpAddress.getLatitude() + "");
+    intent.putExtra(Constant.PICK_UP_LONGITUDE, pickUpAddress.getLongitude() + "");
+    intent.putExtra(Constant.DROP_OFF_LATITUDE, dropOffAddress.getLatitude() + "");
+    intent.putExtra(Constant.DROP_OFF_LONGITUDE, dropOffAddress.getLongitude() + "");
```

Comment:
nit: maybe use Double. toString(d) to convert from a double to a string

Reply:
fixed